packages/core/src/new-api/internal/module-builder.ts

```diff
@@ -163,6 +169,74 @@ export class IgnitionModuleBuilderImplementation<
     return future;
   }
 
+  public call<ContractNameT extends string, FunctionNameT extends string>(
+    contractFuture: ContractFuture<ContractNameT>,
+    functionName: FunctionNameT,
+    args: SolidityParamsType = [],
+    options: CallOptions = {}
+  ): NamedContractCallFuture<ContractNameT, FunctionNameT> {
+    const id = options.id ?? functionName;
+    const futureId = `${this._module.id}:${contractFuture.contractName}:${id}`;
+
+    this._assertUniqueContractId(futureId);
+
+    const future = new NamedContractCallFutureImplementation(
+      futureId,
+      this._module,
+      functionName,
+      contractFuture,
+      args
+    );
+
+    future.dependencies.add(contractFuture);
+
+    for (const arg of args.filter(isFuture)) {
+      future.dependencies.add(arg);
+    }
+
+    for (const afterFuture of (options.after ?? []).filter(isFuture)) {
+      future.dependencies.add(afterFuture);
+    }
+
+    this._module.futures.add(future);
+
+    return future;
+  }
+
+  public staticCall<ContractNameT extends string, FunctionNameT extends string>(
+    contractFuture: ContractFuture<ContractNameT>,
+    functionName: FunctionNameT,
+    args: SolidityParamsType = [],
+    options: CallOptions = {}
+  ): NamedStaticCallFuture<ContractNameT, FunctionNameT> {
+    const id = options.id ?? functionName;
+    const futureId = `${this._module.id}:${contractFuture.contractName}:${id}`;
+
+    this._assertUniqueContractId(futureId);
```

Comment:
Similarly with library, lets create a static call and call version in this PR, then clean them up in a later PR.